src/rdf4cpp/rdf/Graph.cpp

```diff
@@ -78,16 +79,36 @@ bool Graph::serialize(void *const buffer, writer::Cursor *cursor, writer::FlushF
                                                  Variable("o")));
 
     for (auto const &solution : solutions) {
-        RDF4CPP_DETAIL_TRY_SERIALIZE(solution[0]);
-        RDF4CPP_DETAIL_TRY_WRITE_STR(" ");
+        Quad q{solution[0], solution[1], solution[2]};
+        if (!q.serialize_ntriples(buffer, cursor, flush))
+            return false;
+    }
+
+    return true;
+}
+bool Graph::serialize_turtle(void *const buffer, writer::Cursor *cursor, writer::FlushFunc const flush) const noexcept {
```

Comment:
Question: Does this work or will it just basically always be normal ntriples?

Reply:
It works, you get prefixed type IRIs in any case, and if the order in which Triples are serialized is good, you get lists as well.
(There is even a test case for it)

Reply:
Ok yeah, thats what I meant. I would assume that the order is always bad. But I guess you still get some benefits even if its bad